runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -157,40 +160,38 @@ public int hashCode() {
      *         element prevents it from being added to this queue
      */
     public CorfuRecordId enqueue(E e) {
+        checkState(TransactionalContext.isInTransaction(), "must be called within a transaction!");
```

Comment:
I'm concerned, if this change could affect the clients since we changed contract, hope it works

Reply:
Unfortunately there isn't much flexibility here. If clients are enqueuing without a transaction then its incorrect. If we start a transaction internally, then that transaction can abort. They're probably not dealing with the TransacationAbort right now